modin/pandas/dataframe.py

```diff
@@ -594,6 +594,29 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=False):
         )
         return DataFrame(query_compiler=query_compiler)
 
+    def asof(self, where, subset=None):
+        scalar = not is_list_like(where)
+        if isinstance(where, pandas.Index):
+            # Prevent accidental mutation of original:
+            where = where.copy()
+        else:
+            if scalar:
+                where = [where]
+            where = pandas.Index(where)
+
+        df = self
+        if subset is not None:
+            df = self[subset]
+        no_na_index = df.dropna().index
+        new_index = pandas.Index([no_na_index.asof(i) for i in where])
+        result = self.reindex(new_index)
```

Comment:
I prefer `loc` here, `reindex` codepath is not fully optimized yet. (Probably `reindex` should just reuse `loc`).

Reply:
At least on Pandas, `loc` complains that usage with missing labels (in this case, `NaN`) is deprecated. Specifically:

https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#indexing-with-list-with-missing-labels-is-deprecated

Which is why I switched. Do you still want me to switch to `loc`?

Reply:
In fact, it fails:

```
        row_lookup, col_lookup = self._compute_lookup(row_loc, col_loc)
        if any(i == -1 for i in row_lookup) or any(i == -1 for i in col_lookup):
>           raise KeyError(
                "Passing list-likes to .loc or [] with any missing labels is no longer "
                "supported, see https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike"
            )
E           KeyError: 'Passing list-likes to .loc or [] with any missing labels is no longer supported, see https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike'
```